Add documentation about what the representer does. (#670)
exercism/common-lisp: docs/REPRESENTER_NORMALIZATIONS.md

# Representer normalizations

> The [representer][representer] use `[read][cl-read]` to read the submission and `[write][cl-write]` to write the representation as well as applying some additional normalizations.

Author: ```suggestion
The [representer][representer] uses `[read][cl-read]` to read the submission and `[write][cl-write]` to write the representation as well as applying some additional normalizations.
```

In total each submission has the following normalizations applied to it:

## Remove comments

> The submission is `[read][cl-read]` in and so all comments are removed.

Reviewer: I am not sure what is being said here, but it might be:

Maybe it is telling me:

```suggestion
The submission is `[read][cl-read]` in, which removes all comments.
```

Author: I have reworded the sentence- hopefully makes more sense (but does assume a little understanding of `read` which seems reasonable for a document targeting mentors in this language)

## Remove documentation strings

> Instead of normalizing any names which may be present in the documentation string the string is removed. 

Reviewer: There seems to be a comma missing from here, with the prepositional phrase.

I would place the comma(s) but, depending on where I place them, the meaning can change.

Reviewer: Also, why is this file name capitalized compared to other file names?

Reviewer 2: > Also, why is this file name capitalized compared to other file names?

Because all the files within the `docs` directory are capitalized.

Reviewer: Since this is a multi-platform system, should we avoid doing this?  On some platforms case matters, while on others it does not.

However the representation will note if a documentation string was present or not. 

## Normalize text case

> The submission is `[read][cl-read]` in and so all symbols are normalized to uppercase.

Reviewer: The same here, confusion the "and so".

Maybe:

```suggestion
The submission is `[read][cl-read]` in which normalizes symbols to uppercase.
```

Author: I have reworded the sentence- hopefully makes more sense (but does assume a little understanding of `read` which seems reasonable for a document targeting mentors in this language)
